src/amdinfer/workers/rocal.cpp

```diff
@@ -0,0 +1,438 @@
+  }
+  util::Timer timer;
+  timer.add("rocal_batch_start");
+
+  std::vector<unsigned char *> input_batch_buffer; 
+
+  for (unsigned int j = 0; j < batch->size(); j++) {
+      const auto& req = batch->getRequest(j);
+      const auto& inputs = req->getInputs();
+  
+      if (inputs.size() != 1) {
+          req->runCallbackError("Only one input tensor should be present");
+          continue;
+      }
+      const auto& input = inputs.at(0);
+      const auto& input_shape = input.getShape();
+
+      assert(input_shape.size() == 1); // compressed raw buffer should have shape length = 1
+      ROI_xywh_[j].h = input_shape[0];
+      auto input_data = static_cast<unsigned char*>(input.getData());
+      input_batch_buffer.push_back(input_data);
+  }
+
+  bool eos = true; // eos value is always true since there will be only one input tensor 
+  rocalExternalSourceFeedInput(handle_, {}, false, input_batch_buffer, ROI_xywh_, decode_width_, decode_height_, channels_, RocalExternalSourceMode(rocal_external_mode_), 
+                                RocalTensorLayout(0), eos);
+
+  AMDINFER_LOG_INFO(logger, "Beginning rocalRun eval");
+
+  timer.add("eval_start");
+  if (rocalRun(this->handle_) != ROCAL_OK) {
+      AMDINFER_LOG_ERROR(logger, "rocalRun failed to run");
+      throw std::runtime_error("rocalRun failed to run");
+  }
+  timer.add("eval_end");
+
+  auto eval_duration_us = timer.count<std::micro>("eval_start", "eval_end");
+  [[maybe_unused]] auto eval_duration_s = eval_duration_us / std::mega::num;
+  AMDINFER_LOG_INFO(
+    logger,
+    std::string("Finished rocalRun eval; batch size: ") +
+      std::to_string(batch_size_) +
+      "  elapsed time: " + std::to_string(eval_duration_us) +
+      " us.  Images/sec: " + std::to_string(batch_size_ / (eval_duration_s)));
+
+
+  BatchPtr new_batch = batch->propagate();
+  
+  RocalTensorList output_tensor_list = rocalGetOutputTensors(handle_);
+  assert(output_tensor_list->size() == batch->size());
+
+  // RocalTensor output_tensor = output_tensor_list->at(0);
+
+  // unsigned char *out_buffer = nullptr;
+  // float * out_buffer_f = static_cast<float *>output_tensor->buffer();
+  size_t max_decoded_size = 0;
+  for (unsigned int j = 0; j < batch->size(); j++) {
+      const auto & output_tensor = output_tensor_list->at(j);
+      // const int n = output_tensor->dims().at(0);
+      const int h = output_tensor->dims().at(1);
+      const int w = output_tensor->dims().at(2);
+      const int c = output_tensor->dims().at(3);
+      // std::cout << n << " " << c << " " << height << " " << width << " " << std::endl;
+      size_t decoded_size = output_tensor->data_size();
+      if (decoded_size > max_decoded_size) {
+          max_decoded_size = decoded_size;
+      }
+
+      const auto& req = batch->getRequest(j);
+      auto new_request = req->propagate();
+      std::vector<int64_t> shape{h, w, c};
+
+      new_request->addInputTensor(nullptr, shape, amdinfer::DataType::Uint8,
```

Comment:
Yeah this should be fixed to FP32 but it was causing seg fault. Will investigate the issue. 
Does inference server always expect preprocessed input image to be type of FP32?

Reply:
No, the incoming data has a data type attached to it so it could be anything really so you should check the type and if it's not something you can work with, then return an error response

Reply:
I think for input data for rocAL: it is expecting compressed buffer as uchar.

Reply:
This part is actually an output of rocAL and input to the next node(typically inference / training). Therefore, it doesn't necessarily has to be uint8.